src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,536 @@
+    # - direct <--> direct
+    #   - Direct SCRs exist in both Mondo and source
+    in_both_direct_source_edges: Set[RELATIONSHIP] = rels_direct_source_source.intersection(rels_direct_mondo_source)
+    in_both_direct: List[Dict] = _edges_with_metadata_from_plain_edges(in_both_direct_source_edges, source_data_map)
+    #   - Direct SCRs in Mondo which are not direct in source (could be indirect or non-existent)
+    # in_mondo_only_direct_source_edges: Set[RELATIONSHIP] = (
+    #     rels_direct_mondo_source.difference(rels_direct_source_source))
+    # in_mondo_only_direct: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_mondo_only_direct_source_edges, source_data_map)
+    #   - Direct SCRs in source which are not direct in Mondo (could be indirect or non-existent)
+    # in_source_only_direct_source_edges: Set[RELATIONSHIP] = (
+    #     rels_direct_source_source.difference(rels_direct_mondo_source))
+    # in_source_only_direct: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_source_only_direct_source_edges, source_data_map)
+
+    # - indirect <--> indirect
+    #   - Indirect SCRs that exist in both Mondo and source
+    # in_both_indirect_mondo_edges: Set[RELATIONSHIP] = (
+    #     rels_indirect_source_mondo.intersection(rels_indirect_mondo_mondo))
+    # in_both_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(in_both_indirect_mondo_edges, mondo_data_map)
+    #   - Indirect SCRs that exist in source but not in Mondo
+    # in_source_only_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_source_mondo.difference(
+    #     rels_indirect_mondo_mondo)
+    # in_source_only_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_source_only_indirect_mondo_edges, mondo_data_map)
+    #   - Indirect SCRs that exist in Mondo but not in source
+    # in_mondo_only_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_mondo_mondo.difference(
+    #     rels_indirect_source_mondo)
+    # in_mondo_only_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_mondo_only_indirect_mondo_edges, mondo_data_map)
+
+    # - direct <--> ancestor
+    #   - Direct SCRs in source that don't exist at all in Mondo
+    in_source_direct_not_in_mondo_source_edges = rels_direct_source_source.difference(ancestors_mondo_source)
+    in_source_direct_not_in_mondo: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_source_direct_not_in_mondo_source_edges, source_data_map)
+    #   - Direct SCRs in Mondo that don't exist at all in source
+    in_mondo_direct_not_in_source_mondo_edges: Set[RELATIONSHIP] = (
+        rels_direct_mondo_mondo.difference(ancestors_source_mondo))
+    # in_mondo_direct_not_in_source: List[Dict] = _edges_with_metadata_from_plain_edges(
+    #     in_mondo_direct_not_in_source_mondo_edges, mondo_data_map)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
```

Comment:
## Spurious diffs in Case 1 `confirmed` and Case 3 `added-obsolete` @joeflack4 
@matentzn Problem came back. See details.

<details><summary>Details</summary>
<p>
@matentzn This ended up taking a lot more time to clean things up, do some minor bugfixes, and ensure to myself that things are working well. We really need tests for this.

I had fixed the spurious diffs I was seeing. Then I completed "case 5", and then somehow some new spurious diffs came back.

I'm going to set aside time to fix this early on this week. If you want you can wait till after that to merge, but  if you want to merge now, that's fine. I think this is long overdue for merging.

</p>
</details>

Reply:
Was the same sorting problem here, where I previously hadn't accounted for 1 Mondo ID being mapped to 2+ source IDs (proxy merge). I had thought I addressed it but I hadn't yet.